server/src/main/java/org/opensearch/extensions/rest/RestExecuteOnExtensionResponse.java

```diff
@@ -50,33 +39,49 @@ public RestExecuteOnExtensionResponse(RestStatus status, String responseString)
      * @param contentType  The type of the content.
      * @param content  The content.
      * @param headers  The headers.
+     * @param consumedParams  The consumed params.
+     * @param contentConsumed  Whether content was consumed.
      */
-    public RestExecuteOnExtensionResponse(RestStatus status, String contentType, byte[] content, Map<String, List<String>> headers) {
+    public RestExecuteOnExtensionResponse(
+        RestStatus status,
+        String contentType,
+        byte[] content,
+        Map<String, List<String>> headers,
+        List<String> consumedParams,
+        boolean contentConsumed
+    ) {
+        super();
         setStatus(status);
         setContentType(contentType);
         setContent(content);
         setHeaders(headers);
+        setConsumedParams(consumedParams);
```

Comment:
LGTM! Just out of curiosity though, why create all these separate setter methods if they just set fields? Why not just directly set the fields?

Reply:
See the RestSendToExtensionAction class.  This response object is initialized with default settings and that thread blocks until the request handler does something with it, either overwriting it on success, writing failure on exception, or timing out.

Reply:
I realize that didn't really answer the question... basically something outside of this class needs to set these values, thus the setters.  Since they were available I decided to use them here as well.